libraries/chain/backing_store/tests/benchmark_kv.cpp

```diff
@@ -0,0 +1,337 @@
+
+struct dummy_control {
+   fc::logger* get_deep_mind_logger() {
+      return nullptr;
+   }
+
+   uint32_t get_action_id() {
+      return 0;
+   }
+};
+
+struct dummy_context {
+   dummy_control control;
+
+   uint32_t get_action_id() {
+      return 0;
+   }
+};
+
+struct mock_resource_manager {
+   uint64_t billable_size = default_billable_size;
+
+   int64_t update_table_usage(account_name payer, int64_t delta, const kv_resource_trace& trace) {
+      return 0;
+   }
+
+   dummy_context* _context;
+};
+
+using wall_clock_t = std::chrono::time_point<std::chrono::steady_clock>;
+
+// resource_manager, receiver, limits in kv_context are 
+// initialzed as a reference.  Place them globally to avoid 
+// going out of scope
+constexpr uint32_t max_key_size   = 100;
+constexpr uint32_t max_value_size = 1020*1024;
+constexpr uint32_t max_iterators  = 3;
+kv_database_config limits { max_key_size, max_value_size, max_iterators };
+mock_resource_manager resource_manager;
+
+// Main purpose of this program is to microbenchmark individual
+// KV API over RocksDB and ChainKV.
+
+void print_usage(char* program_name) {
+   std::cout << program_name << " - Microbenchmark KV APIs over RocksDb and Chainbase\n\n"
+      << "   -f=FILENAME, file storing keys one per line \n"
+      << "   -o=OPERATION, operation to be benchmarked: get, get_data, set, it_next, it_key_value, Default " << default_operation << std::endl
+      << "   -c, use Chainbase. Without this option: use RocksDb\n"
+      << "   -h, print this help\n"
+      << "   -v=NUMBER, value size. Default " << default_value_size << std::endl
+      << std::endl;
+}
+
+// As this program is only for internal use,
+// we don't use sophisticated library to parse command line
+void parse_cmd_line(int argc, char* argv[], cmd_args& args) {
+   for (auto i = 1; i < argc; ++i) {
+      std::string arg = argv[i];
+      if (arg == "-h") {
+         print_usage(argv[0]);
+         exit(0);
+      } else if (arg == "-f") {
+         if (i + 1 < argc) { 
+            args.key_file = argv[++i];
+         } else {
+            std::cerr << "-f option needs key file name" << std::endl;
+            print_usage(argv[0]);
+            exit(1);
+         }  
+      } else if (arg == "-o") {
+         if (i + 1 < argc) { 
+            args.operation = argv[++i];
+         } else {
+            std::cerr << "-o option needs operation (get, get_data, set)" << std::endl;
+            print_usage(argv[0]);
+            exit(1);
+         }  
+      } else if (arg == "-c") {  // use chainbase
+         args.is_rocksdb = false;
+      } else {
+         std::cout << "Unknown option: " << arg << std::endl;
+         print_usage(argv[0]);
+         exit(1);
+      }
+   }
+}
+
+std::vector<std::string> get_work_set_keys(const cmd_args& args) {
+   std::ifstream key_file(args.key_file);
+   if (!key_file.is_open()) {
+      std::cout << "Failed to open key file " << args.key_file << std::endl;
```

Comment:
Should use `cerr`, not `cout` for error messages.

Reply:
Thanks. Will change.